src/Orleans/Runtime/IGrainActivationContext.cs

```diff
@@ -24,5 +25,9 @@ public interface IGrainActivationContext
 
         /// <summary>Gets a key/value collection that can be used to share data within the scope of the grain activation.</summary>
         IDictionary<object, object> Items { get; }
+
+        IGrainLifeCycle ObservableLifeCycle { get; }
+
+        ParameterInfo[] ConstructorParameters { get; }
```

Comment:
If the activator could somehow provide this info then they could choose their own ctor. Just an idea.